internal/testdata/log_test.go

```diff
@@ -1,85 +0,0 @@
-// Copyright The OpenTelemetry Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//       http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package testdata
-
-import (
-	"testing"
-
-	"github.com/stretchr/testify/assert"
-
-	"go.opentelemetry.io/collector/consumer/pdata"
-	"go.opentelemetry.io/collector/internal"
-	otlpcollectorlog "go.opentelemetry.io/collector/internal/data/protogen/collector/logs/v1"
-)
-
-type logTestCase struct {
-	name string
-	ld   pdata.Logs
-	otlp *otlpcollectorlog.ExportLogsServiceRequest
-}
-
-func generateAllLogTestCases() []logTestCase {
-	return []logTestCase{
-		{
-			name: "one-empty-resource-logs",
-			ld:   GenerateLogsOneEmptyResourceLogs(),
-			otlp: generateLogsOtlpOneEmptyResourceLogs(),
-		},
-		{
-			name: "no-log-records",
-			ld:   GenerateLogsNoLogRecords(),
-			otlp: generateLogOtlpNoLogRecords(),
-		},
-		{
-			name: "one-empty-log-record",
-			ld:   GenerateLogsOneEmptyLogRecord(),
-			otlp: generateLogsOtlpOneEmptyLogRecord(),
-		},
-		{
-			name: "one-log-record-no-resource",
-			ld:   GenerateLogsOneLogRecordNoResource(),
-			otlp: generateLogsOtlpOneLogRecordNoResource(),
-		},
-		{
-			name: "one-log-record",
-			ld:   GenerateLogsOneLogRecord(),
-			otlp: generateLogsOtlpOneLogRecord(),
-		},
-		{
-			name: "two-records-same-resource",
-			ld:   GenerateLogsTwoLogRecordsSameResource(),
-			otlp: generateLogsOtlpTwoLogRecordsSameResource(),
-		},
-		{
-			name: "two-records-same-resource-one-different",
-			ld:   GenerateLogsTwoLogRecordsSameResourceOneDifferent(),
-			otlp: generateLogsOtlpTwoLogRecordsSameResourceOneDifferent(),
-		},
-	}
-}
-
-func TestToFromOtlpLog(t *testing.T) {
```

Comment:
This looks like a useful test that verified LogsFromInternalRep. Why do you want to delete it?

Reply:
LogsFromInternalRep should not be used if we want to extract pdata because we cannot have access to the internal representation. Also I cannot depend on the proto generated files in a non pdata package

Reply:
We do have unit-tests for LogsFromInternalRep, so another reason to not test that functionality here.